File: step07_appsync_with_subscriptions_using_amplify/aws-cdk/lambda/getTodos.ts
const AWS = require('aws-sdk');
const docClient = new AWS.DynamoDB.DocumentClient();

async function getTodos() {
    const params = {
        TableName: process.env.TODOS_TABLE,
    }
    try {
        const data = await docClient.scan(params).promise()
        return data.Items
    } catch (err) {
        console.log('DynamoDB error: ', err)
        return null
    }
}

export default getTodos;

File: step06_appsync_frontend_amplify/backend-cdk/python/functions/getTodos.py
import boto3
import os
import logging
from botocore.exceptions import ClientError

dynamodb = boto3.resource('dynamodb')
tableName = os.environ['TODOS_TABLE']
table = dynamodb.Table(tableName)


def getItem():
    try:
        res = table.scan()
        data = res['Items']
        return data

    except ClientError as e:
        logging.error(e)